src/extensionsIntegrated/CustomSnippets/driver.js

```diff
@@ -87,7 +95,15 @@ define(function (require, exports, module) {
         // update the snippet in the list
         if (snippetIndex !== -1) {
             Global.SnippetHintsList[snippetIndex] = editedData;
-            SnippetsState.saveSnippetsToState();
+
+            // save to file storage
+            SnippetsState.saveSnippetsToState()
+                .then(function () {
+                    //
+                })
+                .catch(function (error) {
+                    console.error("failed to save custom snippet correctly:", error);
```

Comment:
If this should never happen, then report error to bugsnag so that we know exactly why it failed
`logger.reportError(error, "cannot save snippets in on edit snippets`");` . Do this only if its not an expected error, we dont want every exception in bugsnag, only errors we dont expect. 

Also avoid reporting errors that can cause too many entires in bugsnag, what we are particularly interested in in erors dusring app start/setup  , shutdown, module initialization, catastrophic failure etc. normal failures are not usually captured.

Eg: if a user was filling a form and some fields were missing, that is an expected error, and we dont log that. Just check if this specific error should be reported or not.